desc/objectives/_confinement.py

```diff
@@ -0,0 +1,211 @@
+"""Objectives for targeting energy confinement."""
+
+from desc.compute.utils import _compute as compute_fun
+from desc.compute.utils import get_profiles, get_transforms
+from desc.grid import QuadratureGrid
+from desc.utils import Timer, errorif
+
+from .objective_funs import _Objective
+
+
+class HeatingPower(_Objective):
+    """Heating power required by the ISS04 energy confinement time scaling.
+
+    tau_E^ISS04 = 0.134 a^2.28 R^0.64 P^-0.61 n_e^0.54 B^0.84 iota^0.41 (s)
+
+    References
+    ----------
+    https://doi.org/10.1088/0029-5515/45/12/024.
+    Characterization of energy confinement in net-current free plasmas using the
+    extended International Stellarator Database.
+    H. Yamada et al. Nucl. Fusion 29 November 2005; 45 (12): 1684–1693.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f. Defaults to ``target=0``.
+    bounds : tuple of {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f.
+        Defaults to ``target=0``.
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True. Note: Has no effect for this objective.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: Has no effect for this objective.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    f_ren : float, optional
+        ISS04 renormalization factor. Default = 1.
+    H_ISS04 : float, optional
+        ISS04 confinement enhancement factor. Default = 1.
```

Comment:
We could probably combine these into a single term, although I think they are technically separate fudge factors in the literature.

Reply:
I removed `f_ren`